src/Microsoft.DotNet.Build.Tasks/VersionTools/LocalUpdatePublishedVersions.cs

```diff
@@ -5,31 +5,61 @@
 using Microsoft.Build.Framework;
 using Microsoft.DotNet.VersionTools.Automation;
 using System.Diagnostics;
-using System.Linq;
 
 namespace Microsoft.DotNet.Build.Tasks.VersionTools
 {
     public class LocalUpdatePublishedVersions : BuildTask
     {
         [Required]
-        public ITaskItem[] ShippedNuGetPackage { get; set; }
+        public string[] ShippedNuGetPackage { get; set; }
 
         [Required]
         public string VersionsRepoLocalBaseDir { get; set; }
 
         [Required]
         public string VersionsRepoPath { get; set; }
 
+        public string GitHubAuthToken { get; set; }
```

Comment:
I'm not sure I understand why `LocalUpdatePublishedVersions` should talk to GitHub?  What does `Local` mean in this context if it doesn't mean "not online"?

Reply:
I think I get it, we need to get the current versions?  Do we need a Token in that case?  

---
In reply to: [167654618](https://github.com/dotnet/buildtools/pull/1907#discussion_r167654618) [](ancestors = 167654618)

Reply:
I'm taking local to mean the *target* is the local disk, but there may be this optional online source to preserve package versions that aren't being built right now. For source-build, all that matters are the versions produced locally, but when updating dotnet/versions for prodcon, I expect we have the usual servicing scenario.

Token isn't required: anonymous fetching works fine, just with a lower rate limit. (And only the local disk is being updated.)